pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -220,6 +221,16 @@ public boolean setReplicated(boolean replicated) {
 
     @Override
     public CompletableFuture<Void> addConsumer(Consumer consumer) {
+        CompletableFuture<Void> inProgressResetCursorFuture = this.inProgressResetCursorFuture;
+        if (inProgressResetCursorFuture != null) {
+            return inProgressResetCursorFuture.thenCompose(ignore -> addConsumerInternal(consumer))
+                    .exceptionallyCompose(ex -> addConsumerInternal(consumer));
```

Comment:
There's code duplication in this style and the `exceptionallyCompose` cannot differentiate the source of the exception. 
instead of chaining `.thenCompose` and `.exceptionallyCompose`, I think it's better to use `.exceptionally(e -> null).thenCompose(...)` to achieve the intended result of ignoring exceptions.

Reply:
Improved